### PR TITLE
chore(broker-core): message payload is set as variables

### LINKED ISSUE #1614: Message payload is set as local or propagating variables on completion depending on presence of output mappings
Like #1610 but for message catch events (intermediate,  boundary, and start).

### PR DESCRIPTION
**Pull request by saig0**
* store message/job payload in variable state
* message payload is merged/propagated on EVENT_TRIGGERING
* job payload is merged/propagated on ELEMENT_COMPLETING
* output mapping access variable from higher scopes
* new db value for buffers and buffer views

closes #1614

### CODE REVIEW

**Review by npepinpe — COMMENTED** [review]
Looks good, I only had a few comments.

One thing I wasn't sure if I missed or it's not there, do we have a test checking that correlating a boundary event, on payload propagation, we don't affect the attachedTo activity's payload? I think most tests would pass if we were propagating it on the activity's scope which, if it has no input mapping conflicting with the new payload, would the propagate to the workflow scope, and since in most tests we check variable records by name then we might not notice that it was wrongly propagated? That could also be fixed in most tests by not only filtering variable records on their name but also the expected scope key.

File: broker-core/src/main/java/io/zeebe/broker/workflow/processor/event/TriggerBoundaryEventHandler.java [hunk 1]
```diff
@@ -56,11 +58,24 @@ private void triggerInterruptingBoundaryEvent(BpmnStepContext<ExecutableBoundary
 
   private void triggerNonInterruptingBoundaryEvent(
       BpmnStepContext<ExecutableBoundaryEvent> context) {
-    context
-        .getOutput()
-        .appendNewEvent(WorkflowInstanceIntent.EVENT_TRIGGERING, context.getRecord().getValue());
+
```
[the remark this answers is not in the document]
Comment (saig0) [reply broker-core/src/main/java/io/zeebe/broker/workflow/processor/event/TriggerBoundaryEventHandler.java]: The payload copy stuff will be removed with #1899. So I think we don't need to extract this logic here.

File: broker-core/src/main/java/io/zeebe/broker/workflow/processor/flownode/IOMappingHelper.java [hunk 2]
```diff
@@ -18,72 +18,52 @@
...
-      instancePayload = record.getPayload();
+      variablesState.removePayload(elementInstanceKey);
     }
 
-    final DirectBuffer propagatedPayload;
-    if (outputBehavior != ZeebeOutputBehavior.none) {
-      if (element.getOutputBehavior() != ZeebeOutputBehavior.overwrite) {
-        mergeTool.mergeDocument(scopePayload);
-      }
+    if (element.getOutputBehavior() != ZeebeOutputBehavior.none && hasOutputMappings) {
```
[the remark this answers is not in the document]
Comment (saig0) [reply broker-core/src/main/java/io/zeebe/broker/workflow/processor/flownode/IOMappingHelper.java]: The output behavior will be removed with #1878.

**Review by npepinpe — APPROVED** [review]

**Pull request merged**